osutil/disks/labels_test.go

```diff
@@ -72,18 +72,65 @@ func (ts *diskLabelSuite) TestEncodeHexBlkIDFormat(c *C) {
 
 		// these are "unsafe" chars, so they get encoded
 		{"ubuntu data", `ubuntu\x20data`},
-		{"ubuntu\ttab", `ubuntu\x9tab`},
-		{"ubuntu\nnewline", `ubuntu\xanewline`},
+		{"ubuntu\ttab", `ubuntu\x09tab`},
+		{"ubuntu\t9tab", `ubuntu\x099tab`},
+		{"ubuntu\nnewline", `ubuntu\x0anewline`},
 		{"foo bar", `foo\x20bar`},
 		{"foo/bar", `foo\x2fbar`},
 		{"foo/../bar", `foo\x2f..\x2fbar`},
 		{"foo\\bar", `foo\x5cbar`},
 		{"pinkié pie", `pinkié\x20pie`},
 		{"(EFI Boot)", `\x28EFI\x20Boot\x29`},
 		{"[System Boot]", `\x5bSystem\x20Boot\x5d`},
+		// 0x7e is just a 1-rune long character that is not in the allowed set
+		// to demonstrate that these two input strings are encoded/decoded
+		// properly with the constant double width
+		{"ubuntu\x7etab", `ubuntu\x7etab`},
+		{"ubuntu\x07" + "etab", `ubuntu\x07etab`},
+		// works when the only character is an escaped one too
+		{"\t", `\x09`},
 	}
 	for _, t := range tt {
 		c.Logf("tc: %v %q", t.in, t.out)
 		c.Assert(disks.BlkIDEncodeLabel(t.in), Equals, t.out)
+
+		// make sure the other way around works too
+		expin, err := disks.BlkIDDecodeLabel(t.out)
+		c.Assert(err, IsNil)
+
+		c.Assert(expin, Equals, t.in)
+	}
+}
+
+func (ts *diskLabelSuite) TestBlkIDDecodeLabelUnhappy(c *C) {
+	tt := []struct {
+		in     string
+		experr string
+	}{
+		{
+			`\x7z`,
+			"string is malformed, unexpected 'z' character not part of a valid escape sequence",
+		},
+		{
+			`\z`,
+			`string is malformed, unexpected '\\' character not part of a valid escape sequence`,
```

Comment:
and then the error here needs to be updated too

Reply:
mmm but note that this is just how we have to write it for ErrorMatches to work, if we print the error it displays verbatim as:
```
string is malformed, unexpected character '\' not part of a valid escape sequence
```

but you did point out that I can combine all three of these error messages into a single message format instead of redefining it three times